fix(dead-link): update dead autosar link

xmfcx opened the pull request:
Signed-off-by: M. Fatih Cırıt <mfc@leodrive.ai>

## Description

Fixes failing pre-commit-optional ci: https://github.com/autowarefoundation/autoware-documentation/actions/runs/3767529962/jobs/6405151006

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The Reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

xmfcx commented:
I've fixed the dead link. But it is still calling the working link dead. @kenji-miyake

xmfcx commented:
Maybe we should switch to https://github.com/gaurav-nelson/github-action-markdown-link-check instead of the pre-commit plugin?

Current config:
- https://github.com/autowarefoundation/autoware-documentation/blob/main/.pre-commit-config-optional.yaml
- https://github.com/autowarefoundation/autoware-documentation/blob/main/.markdown-link-check.json

xmfcx commented:
More info,
```sh
mfc@mfc-SG13:~$ curl https://www.autosar.org/fileadmin/standards/adaptive/18-03/AUTOSAR_RS_CPP14Guidelines.pdf
curl: (60) SSL certificate problem: unable to get local issuer certificate
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
mfc@mfc-SG13:~$ wget https://www.autosar.org/fileadmin/standards/adaptive/18-03/AUTOSAR_RS_CPP14Guidelines.pdf
--2022-12-29 15:44:59--  https://www.autosar.org/fileadmin/standards/adaptive/18-03/AUTOSAR_RS_CPP14Guidelines.pdf
Resolving www.autosar.org (www.autosar.org)... 195.234.139.22
Connecting to www.autosar.org (www.autosar.org)|195.234.139.22|:443... connected.
ERROR: cannot verify www.autosar.org's certificate, issued by ‘CN=Starfield Secure Certificate Authority - G2,OU=http://certs.starfieldtech.com/repository/,O=Starfield Technologies\\, Inc.,L=Scottsdale,ST=Arizona,C=US’:
  Unable to locally verify the issuer's authority.
To connect to www.autosar.org insecurely, use `--no-check-certificate'.
```

xmfcx commented:
```node
#!/usr/bin/env node

'use strict';

const linkCheck = require('link-check');

linkCheck('https://www.autosar.org/fileadmin/standards/adaptive/18-03/AUTOSAR_RS_CPP14Guidelines.pdf', function (err, result) {
    if (err) {
        console.error(err);
        return;
    }
    console.log(`${result.link} is ${result.status}`);
});
```

This returns `https://www.autosar.org/fileadmin/standards/adaptive/18-03/AUTOSAR_RS_CPP14Guidelines.pdf is dead` too.

xmfcx commented:
After a deep investigation,

- https://github.com/autowarefoundation/autoware-documentation/blob/main/.pre-commit-config-optional.yaml
- https://github.com/autowarefoundation/autoware-documentation/blob/main/.markdown-link-check.json

Through these, 
- [markdown-link-check](https://github.com/tcort/markdown-link-check) uses [link-check](https://github.com/tcort/link-check): https://github.com/tcort/markdown-link-check/blob/master/index.js#L118
- In [this issue](https://github.com/tcort/link-check/issues/30#issuecomment-847786136)
  - > I agree. I can't see a reason why we would ever want to report a link as being good if the TLS checks fail. It's something that needs attention.
    >
    > I've removed the hard coded rejectUnauthorized:false. I'll roll it out in the next version in a few days.
  - It's [removed here](https://github.com/tcort/link-check/commit/d714b878d3ef3977a1d59a513de296b459589db8)
    - With message: "No reason to ever ignore a bad certificate."

So I will find a better link :P

kenji-miyake commented on docs/contributing/coding-guidelines/languages/cpp.md:
```diff
@@ -9,7 +9,7 @@
 
 1. <https://docs.ros.org/en/humble/Contributing/Code-Style-Language-Versions.html>
-2. <https://www.autosar.org/fileadmin/user_upload/standards/adaptive/21-11/AUTOSAR_RS_CPP14Guidelines.pdf>
+2. <https://www.autosar.org/fileadmin/standards/adaptive/18-03/AUTOSAR_RS_CPP14Guidelines.pdf>
```
To point to the latest version.

```suggestion
2. <https://www.autosar.org/fileadmin/standards/adaptive/22-11/AUTOSAR_RS_CPP14Guidelines.pdf>
```

kenji-miyake commented:
@xmfcx Thank you for fixing and investigating this problem! :pray: 
According to https://www.autosar.org/, the website is undergoing maintenance. So I think waiting for a while might solve the problem.
![image](https://user-images.githubusercontent.com/31987104/210100013-8ff82e47-d765-4979-8e2d-3f9229ad510b.png)

kenji-miyake commented:
Or we can temporarily ignore Autosar's URL patterns here.
https://github.com/autowarefoundation/autoware-documentation/blob/a75d24d41699cf739764ad3a60eb83af5c53e6d0/.markdown-link-check.json#L3

xmfcx commented:
@kenji-miyake Thanks, I've updated the link with your suggestion and added the `autosar.org` links to the ignore list. I think it is ready to merge now.
